### PR TITLE
otel: switch to mTLS

### PR DESCRIPTION
**Pull request by frzifus**

### CODE REVIEW

File: observability-hub/otel-collector/otel-collector.yaml [hunk 1]
```diff
@@ -56,11 +54,15 @@ spec:
       otlp:
         protocols:
           grpc:
-            auth:
-              authenticator: basicauth
+            tls: # TODO
+              cert_file: /path/to/server/cert.pem
+              key_file: /path/to/server/key.pem
+              client_ca_file: /path/to/client/ca.pem
```
Comment (frzifus) [inline observability-hub/otel-collector/otel-collector.yaml:60]: Ive to figure out if openshift offers them somewhere or can generate them for us.

**Review by frzifus — COMMENTED** [review]
cc @sallyom

File: observability-hub/otel-collector/otel-collector.yaml (hunk 1 above)
Comment (sallyom) [reply observability-hub/otel-collector/otel-collector.yaml:60]: it does!! I'll get it...

Comment (sallyom) [reply observability-hub/otel-collector/otel-collector.yaml:60]: it's the service-ca-operator & done w/ annotations on configmap, i'll find it

**Pull request merged**